zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -102,9 +102,10 @@ impl SubCircuit<Fr> for MptCircuit<Fr> {
     }
 
     fn min_num_rows_block(block: &witness::Block<Fr>) -> (usize, usize) {
-        // FIXME
         (
-            block.circuits_params.max_mpt_rows,
+            // For an empty storage proof, we may need to lookup the canonical representations of
+            // three different keys. Each lookup requires 32 rows.
+            3 * 32 * block.mpt_updates.len(),
```

Comment:
this number is an upper limit instead of accurate row num?

Reply:
yeah (assuming that the the number of accounts, storage keys in the mpt is less than 2^48). 

a lower limit is `32 * block.mpt_updates.len()`, since each mpt_update has at least 1 key.

Reply:
i think the best way to get an accurate count would be to assign the circuit (but with a mock hash function instead of poseidon). Is that needed? I think in practice there will always be more rows required in the poseidon circuit anyways?

Reply:
Up limit is ok. We need this function be very fast so really assigning is too expensive

Reply:
I remember poseidom round needs 8 row now